src/lib/util/api.ts

```diff
@@ -337,6 +338,16 @@ export async function unfollowUser(id: number) {
     });
 }
 
+// TODO
```

Comment:
Hoping for some help specifically on this part, @IRHM. We need a server-unique but constant identifier here, and a way to fetch the current version. I was hoping you'd have thoughts on how this can be done.

Reply:
@stephaje If the clientIdentifier must be unique for each selfhosted Watcharr instance, then I suppose we could generate one and place it in the watcharr.json config (similar to how we generate the JWT_SECRET in the config) when needed (when plex auth is enabled by an admin). If it only needs to be unique for Watcharr and doesn't matter if different instances share it, we could just hard code a string here.

We have a global variable defined in Vites config: `__WATCHARR_VERSION__` which we use in the face menu, that might work for the version here. If it doesn't, I'm sure it doesn't matter if we just hard code that property at `"1.0.0"`.

Reply:
> (when plex auth is enabled by an admin)

I think it needs to be done immediately so that the admin can use oauth for their own log in. Then it can be disabled if they don't want to use it. The only concern I see here is the possibility that if OAuth breaks or goes down, the admin could be locked out of the account.

Maybe I should write it to support both a base login + associated Plex account?

Reply:
No problem @stephaje,

I guess thats just the downside of using Plex, if their servers go offline, theres not much you can do.

For now, maybe the first admin account should be a normal user, like it is currently. Then they can login via plex to make a new account and give it admin if they'd like.

Then hopefully in the future when we allow linking services to your account we can make this easier for admins.

The confusing part for allowing plex auth and normal auth as backup, is that usernames might not be unique for normal users and plex ones.

If first admin must be a normal user, at least the login can be saved for an emergency. (Then a plex one can be made and given admin if wanted).

Reply:
I'm okay with that strategy. I'll take a look at updating it later today.